src/main/java/com/github/onsdigital/babbage/api/endpoint/content/MaxAge.java

```diff
@@ -54,6 +62,40 @@ protected boolean verifyKey(String key) {
     }
 
     protected int getMaxAge(HttpServletRequest request) throws Exception {
+        if (appConfig().babbage().isLegacyCacheAPIEnabled()) {
+            CloseableHttpClient httpClient = HttpClients.createDefault();
```

Comment:
The contents of this `if` should be moved to its own private function, for clarity. In fact, it could have been also moved into its own separate class, that would have probably made it a lot easier to test, and more readable, and the separation of concerns would have been better, but you don't need to create a new class now if you don't want to, moving it to its own private function should be enough. But you should still split it into smaller functions: for instance, you could have one function to set the headers, another one for all the max-age logic in line 83, etc.